examples/image_classification/image_classification_peft_lora.py

```diff
@@ -0,0 +1,152 @@
+labels = dataset.features["label"].names
+label2id, id2label = dict(), dict()
+for i, label in enumerate(labels):
+    label2id[label] = i
+    id2label[i] = label
+
+# Load image processor
+image_processor = AutoImageProcessor.from_pretrained(args.model_checkpoint)
+
+# Define data transforms
+normalize = Normalize(mean=image_processor.image_mean, std=image_processor.image_std)
+train_transforms = Compose([
+    RandomResizedCrop(image_processor.size["height"]),
+    RandomHorizontalFlip(),
+    ToTensor(),
+    normalize,
+])
+val_transforms = Compose([
+    Resize(image_processor.size["height"]),
+    CenterCrop(image_processor.size["height"]),
+    ToTensor(),
+    normalize,
+])
+
+# Preprocess functions for train and val datasets
+def preprocess_train(example_batch):
+    example_batch["pixel_values"] = [train_transforms(image.convert("RGB")) for image in example_batch["image"]]
+    return example_batch
+
+def preprocess_val(example_batch):
+    example_batch["pixel_values"] = [val_transforms(image.convert("RGB")) for image in example_batch["image"]]
+    return example_batch
+
+# Split the dataset into train and validation sets
+splits = dataset.train_test_split(test_size=0.1)
+train_ds = splits["train"]
+val_ds = splits["test"]
+train_ds.set_transform(preprocess_train)
+val_ds.set_transform(preprocess_val)
+
+# Print trainable parameters of a model
+def print_trainable_parameters(model):
+    trainable_params = 0
+    all_params = 0
+    for _, param in model.named_parameters():
+        all_params += param.numel()
+        if param.requires_grad:
+            trainable_params += param.numel()
+    print(f"trainable params: {trainable_params} || all params: {all_params} || trainable%: {100 * trainable_params / all_params:.2f}")
+
+# Load the base image classification model
+model = AutoModelForImageClassification.from_pretrained(
+    args.model_checkpoint,
+    label2id=label2id,
+    id2label=id2label,
+    ignore_mismatched_sizes=True,
+)
+print_trainable_parameters(model)
+
+# Apply LoRA to the model
+config = LoraConfig(
+    r=16,
+    lora_alpha=16,
+    target_modules=["query", "value"],
+    lora_dropout=0.1,
+    bias="none",
+    modules_to_save=["classifier"],
+)
+lora_model = get_peft_model(model, config)
+print_trainable_parameters(lora_model)
+
+# Define training arguments
+model_name = args.model_checkpoint.split("/")[-1]
+training_args = TrainingArguments(
+    f"{model_name}-finetuned-lora-food101",
+    remove_unused_columns=False,
+    evaluation_strategy="epoch",
+    save_strategy="epoch",
+    learning_rate=args.learning_rate,
+    per_device_train_batch_size=args.batch_size,
+    gradient_accumulation_steps=4,
+    per_device_eval_batch_size=args.batch_size,
+    fp16=True,
+    num_train_epochs=args.num_train_epochs,
+    logging_steps=10,
+    load_best_model_at_end=True,
+    metric_for_best_model="accuracy",
+    push_to_hub=False,
+    label_names=["labels"],
+)
```

Comment:
`output_dir` is not specified in this `training_args`, so the flag `--output_dir` still has no effect